src/bot_driving.py

```diff
@@ -28,6 +31,21 @@ def preprocess(self, img: np.ndarray) -> np.ndarray:
     def postprocess(self, detections: np.ndarray) -> np.ndarray:
         detections = np.clip(detections, -1.0, 1.0)
         return detections
+    
+    def update_buffer(self, steering_signal: tuple[float, float]) -> None:
+        self.buffer[:-1] = self.buffer[1:]
+        self.buffer[-1] = steering_signal
+        
+    def calculate_steering(self) -> tuple[float, float]:
```

Comment:
Mypy raises a warning for the type hints here; `incompatible type "ndarray[Any, Any]"; expected "tuple[float, float]"`

They should work the same way if accessed by the index `[0]` (?)

Reply:
Yeah, I don't use mypy. I think in this case it shouldn't matter because we just expect a pair of two values which can be accessed by their index. We can change that ofc